integration/query_frontend_test.go

```diff
@@ -145,6 +145,14 @@ func runQueryFrontendTest(t *testing.T, setup queryFrontendSetup) {
 		c, err := e2ecortex.NewClient("", queryFrontend.HTTPEndpoint(), "", "", fmt.Sprintf("user-%d", userID))
 		require.NoError(t, err)
 
+		// No need to repeat this test for each user.
+		if userID == 0 && testMissingMetricName {
```

Comment:
[thinking loudly] I'm a bit dubious about adding this further test here. The query-frontend integration test is become the place where everyone adds further cases and we risk to make it just more and more complicated. This case is already covered by the assertion you added to `TestQuerierWithChunksStorage`, so do we really need this too?

Reply:
Originally I suspected that status code is mangled by query-frontend and wanted to make sure that it is propagated properly. Repeating some of the tests between querier and query-frontend makes sense imho, and we should perhaps refactor the code to avoid duplicated code, but repeat those tests.